tests/components/unifiprotect/conftest.py

```diff
@@ -80,6 +81,26 @@ def unifi_dict(self) -> dict[str, Any]:
             "chimes": [c.unifi_dict() for c in self.chimes.values()],
         }
 
+    def get_device_from_mac(self, mac: str) -> ProtectAdoptableDeviceModel | None:
+        """Return device for MAC address."""
+
+        mac = mac.lower().replace(":", "").replace("-", "").replace("_", "")
+
+        all_devices = (
+            self.cameras.values(),
+            self.lights.values(),
+            self.sensors.values(),
+            self.viewers.values(),
+            self.liveviews.values(),
+            self.doorlocks.values(),
+            self.chimes.values(),
+        )
+        for devices in all_devices:
+            for device in devices:
+                if device.mac.lower() == mac:
+                    return device
```

Comment:
I was hoping we could have avoided the linear search for these.

Reply:
Actually it probably doesn't matter unless we start using it during setup as it looks like its only called from `set_chime_paired_doorbells` and I don't think the user is going to call that frequently enough that it matters.

Reply:
That is just for the test because we are mocking the bootstrap, not using the real one. The real one does not do a linear search.

Reply:
It won't matter in that case 👍